Classes/Bookmark/BookmarksViewController.swift

```diff
@@ -0,0 +1,218 @@
+        var cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier)
+        
+        if cell == nil {
+            cell = UITableViewCell.init(style: .subtitle, reuseIdentifier: cellIdentifier)
+        }
+        
+        let bookmark = getBookmarks()[indexPath.row]
+        
+        let titleLabel = "\(bookmark.owner)/\(bookmark.name)"
+        cell?.textLabel?.text = bookmark.type == .repo ? titleLabel : titleLabel + " #\(bookmark.number)"
+        cell?.detailTextLabel?.text = bookmark.title
+        cell?.textLabel?.numberOfLines = 0
+        cell?.detailTextLabel?.numberOfLines = 0
+        cell?.accessibilityTraits |= UIAccessibilityTraitButton
+        cell?.isAccessibilityElement = true
+        cell?.accessibilityLabel = cell?.contentView.subviews
+            .flatMap { $0.accessibilityLabel }
+            .reduce("", { $0 + ".\n" + $1 })
+
+        cell?.imageView?.image = bookmark.getBookmarkImageIcon()
+        cell?.imageView?.tintColor = Styles.Colors.Blue.medium.color
+        return cell!
+    }
+    
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return filterdBookmarks != nil ? false : true // avoid swipe to delete when search bar is active
+    }
+    
+    override func tableView(_ tableView: UITableView,
+                            commit editingStyle: UITableViewCellEditingStyle,
+                            forRowAt indexPath: IndexPath) {
+        if (editingStyle == UITableViewCellEditingStyle.delete) {
+            bookmarkStore.remove(bookmark: getBookmarks()[indexPath.row])
+            tableView.deleteRows(at: [indexPath], with: .automatic)
+        }
+    }
+    
+    // MARK: UITableViewDelegate
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let bookmark = getBookmarks()[indexPath.row]
+        switch bookmark.type {
+        case .repo:
+            let repo = RepositoryDetails(
+                owner: bookmark.owner,
+                name: bookmark.name,
+                hasIssuesEnabled: bookmark.hasIssueEnabled
+            )
+            let repoViewController = RepositoryViewController(client: client, repo: repo)
+            let navigation = UINavigationController(rootViewController: repoViewController)
+            showDetailViewController(navigation, sender: nil)
+
+        case .issue, .pullRequest:
+            let issueModel = IssueDetailsModel(
+                owner: bookmark.owner,
+                repo: bookmark.name,
+                number: bookmark.number
+            )
+            let controller = IssuesViewController(client: client, model: issueModel)
+            let navigation = UINavigationController(rootViewController: controller)
+            showDetailViewController(navigation, sender: nil)
```

Comment:
You can create a `let viewController` and `let navigation` before the switch and the `showDetail` after to avoid some duplication.